docs/apps/packet-io.md

```diff
@@ -0,0 +1,66 @@
+# Packet-IO
+The Packet I/O feature facilitates the exchange of packets between control plane applications and P4 dataplanes. This functionality enables control plane applications to receive packets asynchronously from the dataplane, while also allowing the injection of packets into the dataplane.
+
+The Packet-IO feature is currently supported on the Intel&reg; IPU E2100 target.
+
+## Feature overview
+
+Packet-IO consists of two essential components: Packet-In and Packet-Out.
+- **Packet-In**: Refers to a data plane packet sent by the P4Runtime server to the control plane for further analysis. This is specified as `packetIn` message response in the [p4runtime specification](https://github.com/ipdk-io/p4runtime-dev/blob/mirroring/proto/p4/v1/p4runtime.proto).
+- **Packet-Out**: Defined as a data packet originated by the control plane and injected into the data plane through the P4Runtime server. This is specified as `packetOut` in p4runtime specification.
+
+During the set pipeline sequence, the Packet-IO configuration is extracted from the pipeline configuration. This configuration is then utilized to register RX and TX callbacks with the device driver. When a packet is received, the device driver invokes the RX callback, and when a packet is transmitted, the TX callback is triggered.
+
+### Rx Path
+The P4 device driver invokes the registered RX callback upon receiving a packet, passing the packet details to the Stratum layer of `infrap4d`. The Stratum layer parses the received packet and translates it into a PacketIn message as defined in p4runtime.proto. The P4Runtime server sends the PacketIn message to the connected client.
+
+### Tx Path
+A P4runtime client/controller can send packets to the PacketIO port as a PacketOut message defined in p4runtime.proto. The P4CP Stratum layer translates the PacketOut message to TDI structures and sends them to the driver.
+
+## Enabling Packet-IO
+
+To enable the Packet-IO feature, the `pktio-args` configuration must be added to the following files:
+- The configuration file used by the `infrap4d` process.
+- The configuration file used by `tdi_pipeline_builder`.
+
+The Packet-IO configuration is per device and should be added under the `p4_devices` section.
+
+#### Packet-IO configuration
+	"pktio-args": {
```

Comment:
suggest adding a couple of lines before and after the section to show user where this needs to go in the config file